packages/opentelemetry-core/src/trace/TraceState.ts

```diff
@@ -72,8 +73,13 @@ export class TraceState implements types.TraceState {
       .reduce((agg: Map<string, string>, part: string) => {
         const i = part.indexOf(LIST_MEMBER_KEY_VALUE_SPLITTER);
         if (i !== -1) {
-          // TODO validate key/value constraints defined in the spec
-          agg.set(part.slice(0, i), part.slice(i + 1, part.length));
+          const key = part.slice(0, i);
+          const value = part.slice(i + 1, part.length);
+          if (validateKey(key) && validateValue(value)) {
```

Comment:
There should be an additional test to ensure this is used on the correct key/value.

Reply:
Added more tests.